src/wp-includes/pluggable.php

```diff
@@ -351,6 +351,9 @@ function wp_mail( $to, $subject, $message, $headers = '', $attachments = array()
 								} elseif ( false !== stripos( $charset_content, 'boundary=' ) ) {
 									$boundary = trim( str_replace( array( 'BOUNDARY=', 'boundary=', '"' ), '', $charset_content ) );
 									$charset  = '';
+									if ( preg_match( '~multipart/([a-z]+)~i', $content_type, $matches ) ) {
```

Comment:
While this handles `multipart/alternative` and `multipart/byteranges`, what about the hypothetical `multipart/form-data`? I think it would be safer to include all characters up until whitespace or `;`: 

```suggestion
									if ( preg_match( '~multipart/([^\s;]+)~', $content_type, $matches ) ) {
```

Reply:
Or actually, there shouldn't be a `;` at all since the content was exploded already, so this can be:


```suggestion
									if ( preg_match( '~^multipart/(\S+)~', $content_type, $matches ) ) {
```

Also adding the anchor to check that `multipart` appears at the beginning of the string.

Reply:
@westonruter this feels more like an improvement to me (we are moving from a total regression that happened years ago and no one took the time to solve it, to let's sort all cases even the most random ones). 

But some testing might be done, and with the current build plenty of testing has already been done, which gives me more confidence currently, and we are a little tight on time right now. Wouldn't it be better to get this PR pushed and thereafter create a new ticket with this little tweak and get some testing going there and push it lately?

Or are you 100% sure that this regex is going to work well, and I push it blindly?

On the bright part, at least unit tests are passing

Reply:
Yes, the pattern in 1fd5ac9260411f9c8fe9fcb26429b1888c54be9f seems more robust to me. It's ensuring the `$content_type` _starts_ with `multipart/` and that the match group is all following sequential non-whitespace characters.

Reply:
@westonruter ok, I have not tested nor reviewed that pattern. If you feel confident with it, let's shoot this ball.

Devil loads regex 😅